_sections/93-compact-representation.md

```diff
@@ -30,31 +30,36 @@ Given the following ODK XForm definition:
 Full records might look like:
 
 {% highlight xml %}
-<household id="household_survey" orx:version="2018061801" odk:prefix="hh" odk:delimiter="|">
+<household id="household_survey" orx:version="2018061801" odk:prefix="hh" odk:delimiter="+">
 	<meta>
 		<instanceID>uuid:82724cc5-df6f-46bf-86d5-26683ae35d5b</instanceID>
 	</meta>
-	<lastname odk:tag="ln">Bar</lastname>
-	<age>10</age>
+	<person>
+		<firstname odk:tag="fn" />
+		<lastname odk:tag="ln">Bar</lastname>
+		<age>10</age>
+	</person>
 </household>
 {% endhighlight %}
 
 {% highlight xml %}
-<household id="household_survey" orx:version="2018061801" odk:prefix="hh" odk:delimiter="|">
+<household id="household_survey" orx:version="2018061801" odk:prefix="hh" odk:delimiter="+">
 	<meta>
 		<instanceID>uuid:82724cc5-df6f-46bf-86d5-26683ae35d5b</instanceID>
 	</meta>
-	<firstname odk:tag="fn">Mary Kate</firstname>
-	<lastname odk:tag="ln">Doe</lastname>
-	<age>15</age>
+	<person>
+		<firstname odk:tag="fn">Mary Kate</firstname>
+		<lastname odk:tag="ln">Doe</lastname>
+		<age>15</age>
+	</person>
 </household>
 {% endhighlight %}
 
 The compact representations of those records would be:
-`hh|fn||ln|Bar`
+`hh+ln+Bar`
 
-`hh|fn|Mary Kate|ln|Doe`
+`hh+fn+Mary Kate+ln+Doe`
 
-If the delimiter is included in one of the question values, it will be prepended by a slash. For example, the first name `"Mary|Kate"` would be represented as `"Mary\|Kate"`
+If the delimiter is included in one of the question values, it will be prepended by a slash. For example, the first name `"Mary Kate"` would be represented as `"Mary\ Kate"` if the default space delimiter is used.
 
-As in the regular representation, nodes that are not relevant are not included in the compact representation, even if the `tag` attribute is defined.
\ No newline at end of file
+As in the regular representation, nodes that are not relevant are not included in the compact representation. Unlike in the regular representation, nodes that are relevant but empty are not included in the compact representation, even if they have an `odk:tag`.
```

Comment:
Just checking. So the argument is that the loss of having a distinction between those relevant empty nodes and irrelevant nodes is worth the gain in having a smaller submission size, right?

Reply:
Yes, exactly. Should have called that out explicitly in the description. I agree that it's a worthwhile distinction for the full record. As far as I know, this is only used for SMS messages at the moment and they're almost never meant to stand alone. Small size really matters.

I suppose that decision is still not set in stone and could be changed with minimal impact if you think it's not a good way to go. I can't remember whether we explicitly discussed it. And bizarrely I wrote the original example exactly backwards by not including the node in the "normal" submission and including it in the compact submission. I don't know what that was about.

Reply:
(It can be changed with minimal impact because either users are parsing the results by looking for the tag names or they're using position in which case they expect every node to be populated and hopefully are making the questions required)

Reply:
I think that's fine. Thanks. Medic Mobile is also not including them in their implementation as @alxndrsn mentions [here](https://github.com/enketo/enketo-core/issues/570#issuecomment-432732607), so let's merge!

Reply:
That comment of mine's a little misleading because:
1. I was a little confused at the time between irrelevant fields and empty fields
2. we're not actually doing this in production yet, and our implementation will depend on what ends up in the spec!